swarm/src/behaviour.rs

```diff
@@ -449,10 +439,12 @@ pub enum FromSwarm<'a, Handler> {
     ListenerError(ListenerError<'a>),
     /// Informs the behaviour that a listener closed.
     ListenerClosed(ListenerClosed<'a>),
-    /// Informs the behaviour that we have discovered a new external address for us.
-    NewExternalAddr(NewExternalAddr<'a>),
+    /// Informs the behaviour that we have discovered a new candidate for an external address for us.
+    NewExternalAddrCandidate(NewExternalAddrCandidate<'a>),
     /// Informs the behaviour that an external address was removed.
```

Comment:
```suggestion
    /// Informs the behaviour that an external address of the local node was removed.
```

Just to avoid confusion with external addresses of remote nodes.